source/UF/TruncationLevels.lagda

```diff
@@ -79,3 +80,72 @@ instance
  _≤_ {{Order-ℕ₋₂-ℕ₋₂}} = _≤ℕ₋₂_
 
 \end{code}
+
+Added by Ian Ray 22nd September, 2024.
+
+We show
+ ℕ₋₂ ≃ ℕ
+
+\begin{code}
+
+ℕ₋₂-to-ℕ : ℕ₋₂ → ℕ
+ℕ₋₂-to-ℕ −2 = 0
+ℕ₋₂-to-ℕ (succ x) = succ (ℕ₋₂-to-ℕ x)
+
+ℕ-to-ℕ₋₂' : ℕ → ℕ₋₂
+ℕ-to-ℕ₋₂' 0 = −2
+ℕ-to-ℕ₋₂' (succ x) = succ (ℕ-to-ℕ₋₂' x)
+
+ℕ₋₂-ℕ-equivalence : ℕ₋₂ ≃ ℕ
+ℕ₋₂-ℕ-equivalence =
+ (ℕ₋₂-to-ℕ , qinvs-are-equivs ℕ₋₂-to-ℕ (ℕ-to-ℕ₋₂' , H , G))
+ where
+  H : ℕ-to-ℕ₋₂' ∘ ℕ₋₂-to-ℕ  ∼ id
+  H −2 = refl
+  H (succ x) = ap succ (H x)
+  G : ℕ₋₂-to-ℕ ∘ ℕ-to-ℕ₋₂' ∼ id
+  G 0 = refl
+  G (succ x) = ap succ (G x)
+
+\end{code}
+
+We demonstrate an analogous notion of 'subtraction' in ℕ₋₂.
+
+\begin{code}
+
+ℕ₋₂-to-ℕ' : ℕ₋₂ → ℕ
+ℕ₋₂-to-ℕ' −2 = 0
+ℕ₋₂-to-ℕ' (succ −2) = 0
+ℕ₋₂-to-ℕ' (succ (succ −2)) = 0
+ℕ₋₂-to-ℕ' (succ (succ (succ x))) = succ (ℕ₋₂-to-ℕ' (succ (succ x)))
+
+telescoping-sum-2 : (n : ℕ₋₂) → (−2 + ℕ₋₂-to-ℕ' (succ (succ n))) ＝ n
+telescoping-sum-2 −2 = refl
+telescoping-sum-2 (succ n) = ap succ (telescoping-sum-2 n)
+
+succ-ℕ₋₂-assoc : (m : ℕ₋₂) (n : ℕ) → succ m + n ＝ succ(m + n)
+succ-ℕ₋₂-assoc m 0 = refl
+succ-ℕ₋₂-assoc m (succ n) = ap succ (succ-ℕ₋₂-assoc m n)
+
+subtraction-ℕ₋₂ : (m n : ℕ₋₂) → m ≤ℕ₋₂ n → Σ k ꞉ ℕ , m + k ＝ n
+subtraction-ℕ₋₂ −2 n o = (ℕ₋₂-to-ℕ' (n + 2) , telescoping-sum-2 n)
+subtraction-ℕ₋₂ (succ m) (succ n) o = (k , p)
+ where
+  IH : Σ k ꞉ ℕ , m + k ＝ n
+  IH = subtraction-ℕ₋₂ m n o
+  k = pr₁ IH
+  q = pr₂ IH 
+  p : (succ m + k) ＝ succ n
+  p = succ m + k ＝⟨ succ-ℕ₋₂-assoc m k ⟩
+      succ(m + k)＝⟨ ap succ q ⟩
+      succ n     ∎
+
+subtraction-ℕ₋₂-term : (m n : ℕ₋₂) → m ≤ℕ₋₂ n → ℕ
```

Comment:
And here.

Reply:
Okay take a look now. Also would you like me to use it when I am defining something in line where it is used in patternmatching. For example see the definition of succ-ℕ₋₂-assoc